### DISCUSSION TITLE
Min und Max für Temperatursteuerung aus CCU nach HA übernehmen? Mind 5 Grad statt 4,5 Grad?

### DISCUSSION THREAD

**Discussion by Bascht74** [SukramJ/aiohomematic#661] [Ideas]
**Is your feature request related to a problem? Please describe.**
no

**Describe the solution you'd like**
Hallo, erst nocheinmal vielen dank für diese tolle Custom-Integratoin für HM(IP), um diese Welt mit der von HA zu verbinden!
Ich benutze in der CCU3 die Heizungsstuerung und habe dazu mehrere Thermostate an der Wand oder an der Heizung installiert. Einige davon sind auch in Geschäftsräumen installiert, bei denen sich die Temperatur nur innerhalb einer bestimmten Range verändern lässt. Diese Range habe ich in der CCU festgelegt, z.B. 14,5 Grad / 22 Grad:
![image](https://user-images.githubusercontent.com/66269310/196820477-f3003875-43b2-4b6e-a93d-823d4d2e932a.png)

Nur wird diese Beschränkung derzeit nicht nach HA übernommen. Das bedeutet, dass man die Temperaturen zwischen 4,5 und 30,5 Grad einstellen kann und dann über die CCU wieder ein gültiger Wert zurückgegeben wird. 
Weiterhin kann man die Temperatur auf 4,5 Grad setzen und danach wird das Gerät von Heizen auf aus gesetzt.

Sofern es möglich ist, wäre es toll, dass dieser festgelegte Werterahmen auch in HA berücksichtigt/angezeigt werden könnte. 
Im Thermostat gibt es eine solche möglichkeit (siehe: https://developers.home-assistant.io/docs/core/entity/climate/ ):

<html>
<body>

max_temp | float | DEFAULT_MAX_TEMP (value == 35) | Returns the maximum temperature.
-- | -- | -- | --
min_temp | float | DEFAULT_MIN_TEMP (value == 7) | Returns the minimum temperature.

</body>
</html>

Auf jeden Fall wäre es m.E. sinnvoll, in HA die Temperatureinstellung auf 5 Grad zu beschränken, damit nicht bei der Einstellung auf 4,5 Grad sich der Zustand des Geräts von Heizen auf "aus" ändert. Wenn ich das Gerät ausschalten möchte, sollte ich das in HA über die vorgesehene Möglichkeit tun und nicht über das setzen einer bestimmten Temperatur.

**Describe alternatives you've considered**
./.

**Additional context**
Ich habe mir versucht, einmal die API anzuschauen und bin mir nicht sicher, ob man an diese Konfigurationsdaten herankommt.

Schön wäre es wie gesagt auf jeden Fall, wenn die untere Temperatur nicht auf mind. 4,5 sondern auf mind. 5,0 Grad  (oder einen höheren Wert wie in der CCU eingestellt) eingestellt werden kann. 
Die Möglichkeit zum Setzen von 4,5 Grad in HA sollte deaktiviert werden, da sie bei keinem (zumindest mir bekanntn) anderen Heizungsgerät in HA ein derartiges Verhalten zeigt.

In Summe finde ich übrigens die Integration mehr als gelungen! Ein echtes Highlight bei den HACS-Integrationen.

**Comment by Bascht74** [comment]
Oh, jetzt habe ich meinen Eintrag wiedergefunden :-) 
Hatte schon gedacht, dass ich nicht richtig auf speichern gedrückt habe. 
Bin gespannt und freue mich auf ein Feedback!

**Comment by ThomDietrich** [comment]
Hallo @Bascht74,
ich beziehe mich nur auf einen Teil deiner Nachricht, zu deinem Hauptthema werden sich mit Sicherheit andere melden.

> Auf jeden Fall wäre es m.E. sinnvoll, in HA die Temperatureinstellung auf 5 Grad zu beschränken, damit nicht bei der Einstellung auf 4,5 Grad sich der Zustand des Geräts von Heizen auf "aus" ändert. Wenn ich das Gerät ausschalten möchte, sollte ich das in HA über die vorgesehene Möglichkeit tun und nicht über das setzen einer bestimmten Temperatur.

Ich stimme dir total zu. Ich selbst nutze den 4,5°C workaround noch aus historischen Gründen, das sollte allerdings mittlerweile nicht mehr benötigt und daher aus Nutzerfreundlichkeits-Gründen auch nicht mehr angeboten werden.

**Reply by Bascht74** [reply]
Oh, diese Historie kenne ich gar nicht. Ich bin neu dabei uns sehe nur, wie sich das aktuell darstellt.

**Comment by Baxxy13** [comment]
Die in der WebUI für's Thermostat gespeicherte "Minimale Temperatur" und "Maximale Temperatur" liegt im MASTER_PARAMSET.
Bei IP normalerweise in Kanal:1 bei HM direkt im Gerät. 
Ich "glaube" nicht das davon irgendwas für diese Integration ausgelesen und genutzt wird. 
"Machbar" wäre es bestimmt.

Ich soll zwar nicht immer mit der WebUI vergleichen, mache es aber trotzdem.
Auch da kann man bspw. erstmal einen geringeren Wert als "Minimale Temperatur" übergeben welcher aber nicht vom Gerät übernommen wird und auf die "Minimale Temperatur" (zurück)springt. Ist jetzt mittels Integration erstmal nicht anders.

Das mit dem "AUS" bei 4,5°C kommt ja auch aus der WebUI. Nur wenn die "Minimale Temperatur" in den Geräteeinstellungen auf "AUS" steht bekommt man das "AUS" beim Gerät zu Gesicht. 
Die Frage wäre: Ist es dann wirklich aus, stellt also jegliche Regelung ein?


> Wenn ich das Gerät ausschalten möchte, sollte ich das in HA über die vorgesehene Möglichkeit tun und nicht über das setzen einer bestimmten Temperatur.

Wie sollte denn sowas ablaufen? HM/IP Thermostate kennen doch gar kein echtes AUS. 
AUS bei HM/IP ist nun mal gleichbedeutend mit MANU/4,5°C

**Comment by Bascht74** [comment]
Hallo Baxxy13, 

erstmal Danke für Deine Gedanken. 

Ich selbst habe hier einfach keinen Vergleich gemacht, sondern geschaut, was Home Assistant üblicherweise anbietet und generell möglich wäre. Die CCU benutze ich nur zur (Grund-)Konfiguration und beim Look und Feel ist HA ggü. der CCU schon ein ganzes Stück weiter. Es ist schonmal schön zu wissen, dass die Daten über den Kanal:1 verfügbar wären.

Optisch sehe ich auch in der CCU "Aus" in der Benutzeroberfläche:
![image](https://user-images.githubusercontent.com/66269310/197003201-1f5066e7-6c09-4da0-8e67-d993251271e1.png)
Die 4,5 Grad scheinen wie Du schreibst wohl eher intern verwendet zu werden und der Benutzer sieht das nicht. 

Da auch die CCU die manuelle Einstellung von 4,5 Grad in der Oberfläche nicht anbietet, dann sollte man in HA m.E. nicht auf 4,5 Grad stellen können. Scheinbar ist das "aus" in HA (wenn ich den Eintrag von @ThomDietrich richtig verstehe)  vor einiger Zeit dazu gekommen und die 4,5 Grad noch eine "historische Altlast". 

In HA ist das Verhalten für Benutzer, die eher HA kennen jetzt eigenartig: 
Wenn ich auf 5 Grad stelle, dann ändert sich die Temperatur und beim nächsten Zeitpunkt wird der Automatik-Betrieb fortgesetzt. Stelle ich aber ganz nach links (4,5 Grad = vermeintliche Mindesttemperatur), dann wird plötzlich von Automatikbetrieb auf manuell gewechselt und die Heizung ausgeschaltet. Ein halbes Grad ändert die komplette Logik, ohne dass man das erkennen kann (muss man wissen)
Aus diesem Grund habe ich mich entschieden diesen Feature Request überhaupt zu schreiben.

Jeder, der von Grundauf die CCU und die Eigenarten von HM kennt, sagt vielleicht "weiß ich doch". Aber für neue Einsteiger / Kinder / ältere Menschen (gerade installiert) usw. wäre das eine Hilfe, wenn sich das HA Thermostat so nicht verhalten würde. 
Dazu müsste nur der statische untere Mindestwert auf mind. 5 Grad geändert werden (oder - falls das möglich ist auf die Mind. Temp. aus Kanal1).

Ich hoffe, ich konnte nochmal verdeutlichen, worum es mir aus der Sicht eines reinen "Users" hier von der "User Experiance" her geht.

Gruß

Sebastian

**Reply by Baxxy13** [reply]
Ahh, jetzt verstehe ich (ein bisschen) das "Problem".
Komme aber nicht auf die gleichen Erkenntnisse wie du.
Außerdem gibt es (wie üblich) Unterschiede zwischen HM und IP Geräten.

HM: 
Das HM-Thermostat lässt sich im AUTO-MODUS auf 4,5°C herunterregeln und zeigt dann AUS. 
Es bleibt aber auf AUTO und springt demzufolge beim nächsten Wochenprogramm-Schaltzeitpunkt auf den dort eingestellten Wert zurück. 
Wie sollte das Verhalten bei HM sein?
![GIF 20 10 2022 19-35-49](https://user-images.githubusercontent.com/54898249/197020540-e0adf493-1b0b-4090-9e90-a13f14218a2c.gif)

IP:
Das IP-Thermostat lässt sich im AUTO-MODUS auf 4,5°C herunterregeln, reagiert aber nicht darauf.
Somit wäre die Sollwertanzeige in HA (vermutlich) bis zur nächsten zyklischen Datenübertragung falsch.
Wie sollte das Verhalten bei IP sein?
![GIF 20 10 2022 19-37-39](https://user-images.githubusercontent.com/54898249/197021340-16dff48c-e538-4eca-bef3-4d0ada3c9863.gif)

Aus dem Bauch heraus würde ich jetzt auch sagen das die Slider auf minimum 5°C begrenzt sein müssten (oder halt auf die "Minimale Temperatur" aus den Geräteeinstellungen). 
Der wählbare "AUS"-Modus müsste aber weiterhin MANU + 4,5°C setzen. 
Das wiederum geht aber nur wenn die "Minimale Temperatur" in den Geräteeinstellungen auch auf "AUS" steht.
Oder wie sollte man sonst das "AUS" der Thermostate definieren?

**Reply by SukramJ** [reply]
Ich denke 5 Grad und Min/Max kann man machen

**Reply by SukramJ** [reply]
Die Umsetzung könnt ihr mit 1.19.7 testen.

**Reply by Bascht74** [reply]
@Baxxy13 Ich habe bei mir nochmal getestet. Bei mir geht das Gerät bei 4,5 auf "aus", also reagiert auch darauf. Es ist aber ein HmIP-eTRV-2  und nicht der Luft und Feuchtigkeitsensor STHD...

**Reply by Bascht74** [reply]
@SukramJ Danke erstmal für die Umsetzung von "min 5°". Das funktioniert jetzt auch prima.
Allerdings werden die Min/Max-Werte bei mir nicht übernommen in HA. 
Im Commit habe ich dazu auch nichts (als Laie) gefunden: 
https://github.com/danielperna84/custom_homematic/pull/337/files
Fehlt hier evtl. noch ein Commit?

**Reply by SukramJ** [reply]
Ne, du must noch den Service homematicip_local.clear_cache ausführen, und HA danach neustarten.

**Reply by Bascht74** [reply]
Danke für den Hinweis. Funktioniert jetzt perfekt! Vielen dank! Jetzt kann man das viel besser einstellen, wenn man diese Werte definiert hat. 
Leider geht ja kein MIN-Wert in Verbindung mit aus (der Min muss dann ja auf "AUS" stehen, ansonsten kann man nicht ausschalten).

**Reply by Baxxy13** [reply]
Sieht gut aus! 
Muss aber auch zugeben das ich bis zum heutigen Tag noch nie an einem der Slider gedreht hatte, läuft halt alles auf Automatik.

Das "AUS" nun nicht mehr geht wenn "Minimale Temperatur" in den Geräteeinstellungen auf irgendwas anderes als "Aus" gestellt ist hatte ich ja schon vorher geschrieben.
